public/app/features/dashboard/components/ShareModal/SharePublicDashboard/ConfigPublicDashboard/Configuration.tsx

```diff
@@ -19,21 +15,16 @@ export const Configuration = ({
   disabled,
   onChange,
   register,
+  timeRange,
```

Comment:
something to think about is if ConfigPublicDashboard is responsible for passing the timeRange to his child, just because he is using dashboard from `dashboard.getModel() `for another purpose. 
Why not call `dashboard.getModel() again ` in Configuration component as it was and let this component know how to get the timeRange?
 
I don't know, just something to think about.

Reply:
For me timeRange makes sense to be passed down to `Configuration` because `Configuration` is a rendering component, it is not concerned with operations - just listing form inputs. Also now it is a sibling of `SettingsSummary` which needs to render some of the same information. Since both require `timeRange`, we'd have to call 3 more functions in each of them to get the same timeRange.
If we ever change where we get the timeRange from or even what a dashboard is, it would be good if it's not a concern of the rendering components, but of pages or parents. That's at least in my opinion, otherwise I can also see why Configuration can have a separate life, but now that its sibling renders the same information, I took the responsibility up a level.

Reply:
Yeah, maybe now it makes sense to pop up the info, since it's sibling also needs it.  Great